lib/src/country.dart

```diff
@@ -119,14 +119,30 @@ class Country {
     return data;
   }
 
-  bool startsWith(String query, CountryLocalizations? localizations) =>
-      name.toLowerCase().startsWith(query.toLowerCase()) ||
-      countryCode.toLowerCase().startsWith(query.toLowerCase()) ||
-      (localizations
-              ?.countryName(countryCode: countryCode)
-              ?.toLowerCase()
-              .startsWith(query.toLowerCase()) ??
-          false);
+  bool startsWith(String query, CountryLocalizations? localizations) {
+    if(query.startsWith("+")){
+      final String queryWithoutPlusSign = query.replaceAll("+", "").trim();
+      return
+        phoneCode.startsWith(queryWithoutPlusSign.toLowerCase()) ||
+            name.toLowerCase().startsWith(queryWithoutPlusSign.toLowerCase()) ||
+            countryCode.toLowerCase().startsWith(queryWithoutPlusSign.toLowerCase()) ||
+            (localizations
+                ?.countryName(countryCode: countryCode)
+                ?.toLowerCase()
+                .startsWith(queryWithoutPlusSign.toLowerCase()) ??
+                false);
+    }else{
+      return
+        phoneCode.startsWith(query.toLowerCase()) ||
+            name.toLowerCase().startsWith(query.toLowerCase()) ||
+            countryCode.toLowerCase().startsWith(query.toLowerCase()) ||
+            (localizations
+                ?.countryName(countryCode: countryCode)
+                ?.toLowerCase()
+                .startsWith(query.toLowerCase()) ??
+                false);
+    }
+  }
```

Comment:
Please simplified. You wrote the "Search condition" twice. 
```suggestion
  bool startsWith(String query, CountryLocalizations? localizations) {
    String _query = query;
    if (query.startsWith("+")) {
      _query = query.replaceAll("+", "").trim();
    }

      return phoneCode.startsWith(_query.toLowerCase()) ||
          name.toLowerCase().startsWith(_query.toLowerCase()) ||
          countryCode.toLowerCase().startsWith(_query.toLowerCase()) ||
          (localizations
              ?.countryName(countryCode: countryCode)
              ?.toLowerCase()
              .startsWith(_query.toLowerCase()) ??
              false);
    }
```